File: Data Structures & Algorithms/merge-triplets-to-form-target/submission-0.py
class Solution:
    def mergeTriplets(self, triplets: List[List[int]], target: List[int]) -> bool:
        # find a subset so that target is maximum of each
        found = [0,0,0]
        for t in triplets:
            if t[0]>target[0] or t[1]>target[1] or t[2]>target[2]:
                continue
            for i in range(3):
                if t[i]==target[i]:
                    found[i]=True
        return bool(found[0] and found[1] and found[2])
            

File: Data Structures & Algorithms/valid-sudoku/submission-0.cpp
class Solution {
public:
    bool isValidSudoku(vector<vector<char>>& board) {
        int m = board.size(); int n = board[0].size();
        for (int i=0; i<m; i++){
            vector<int> found(n);
            for (int j=0; j<n; j++){
                if (board[i][j]!='.') found[board[i][j]-'1']++;
            }
            for (int b: found){
                if (b>1) return false;
            }
        } 
        for (int j=0; j<n; j++){
            vector<int> found(m);
            for (int i=0; i<m; i++){
                if (board[i][j]!='.') found[board[i][j]-'1']++;
            }
            for (int b: found){
                if (b>1) return false;
            }
        }
        for (int i=0; i<m; i+=3){
            for (int j=0; j<n; j+=3){
                vector<int> found(m);
                for (int k=i; k<i+3; k++){
                    for (int l=j; l<j+3; l++)
                        if (board[k][l]!='.') found[board[k][l]-'1']++;
                }
                for (int b: found){
                    if (b>1) return false;
                }
            }
        }

        return true;
    }
};
